QbQuestionsAPI/Persistence/Repositories/QbQuestionRepository.cs

```diff
@@ -26,10 +26,24 @@ public async Task<QbQuestion> FindByIdAsync(int id)
         public async Task<QbQuestion> GetRandomAsync(int? level)
         {
             #pragma warning disable EF1000
-            string whereClause = level == null ? string.Empty : $"WHERE Level = {level}";
-            // TODO: Consider using SQL parameter for whereClause
+            // TODO: Consider using SQL parameters
+            int count = level == null
+                ? await _context.QbQuestions.CountAsync(q => q.Id > 0)
+                : await _context.QbQuestions.CountAsync(q => (int)q.Level == level);
+            Random randomGenerator = new Random();
+            int random = randomGenerator.Next() % count;
+
+            string levelCondition = level == null ? string.Empty : $"WHERE Level = {level}";
+            string innerQuery = $@"(SELECT ROW_NUMBER() OVER (ORDER BY (SELECT 1))
+                                    AS RowNumber, *
```

Comment:
`ROW_NUMBER()` requires an `ORDER BY` clause, the `SELECT 1` here should get optimized out, so no sorting.